ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -498,6 +539,8 @@ data ApplyMapKind :: MapKind -> Type -> Type -> Type where
   ApplyTrackingMK :: !(UtxoValues  k v) -> !(UtxoDiff k v) -> ApplyMapKind TrackingMK   k v
   ApplyValuesMK   :: !(UtxoValues  k v)                    -> ApplyMapKind ValuesMK     k v
   ApplyRewoundMK  :: !(RewoundKeys k v)                    -> ApplyMapKind RewoundMK    k v
+  ApplyLMDBMK     :: !String -> !(LMDB.Database k v)       -> ApplyMapKind LMDBMK       k v
```

Comment:
Hmm. We probably can't serialize this, so the closed instance for `ApplyMapKind` will probably need to be divided into several open instances now. 😬

Reply:
Are you sure it needs to be serialised?

Reply:
Right now, the in-memory back-end serialises a `ValuesMK`. But we could probably arrange it so that we don't propagate a `ToCBOR (ApplyMapKind ...)` constraint of any kind.

In which case, yes, no instances open-or-closed needed for serializing `ApplyMapKind`